var/spack/repos/builtin/packages/plumed/package.py

```diff
@@ -292,10 +292,9 @@ def configure_args(self):
         if "+mpi" in spec:
             configure_opts.extend(["--enable-mpi", "CXX={0}".format(spec["mpi"].mpicxx)])
 
-            # If the MPI dependency is provided by the intel-mpi package then
-            # the following additional argument is required to allow it to
-            # build.
-            if "intel-mpi" in spec:
+            # If the MPI dependency is provided by the intel-oneapi-mpi package then the following
+            # additional argument is required to allow it to build.
+            if spec.satisfies("^[virtuals=mpi] intel-oneapi-mpi"):
```

Comment:
shouldn't this conditional have an OR logic, to include both cases of intel-oneapi-mpi and intel-mpi?

as in
```
if "intel-mpi" in spec or spec.satisfies("^[virtuals=mpi] intel-oneapi-mpi"):
```

Reply:
@marcodelapierre The point is that I'm trying to remove `intel-mpi`, `intel-mkl` and `intel-parallel-studio` in the PR linked in the description, in favor of `intel-oneapi-*` counterparts.

Reply:
Makes sense then! 
 
Unfortunately I don't have the chance to test it, but looks sensible.  
I have checked in the plumed repo, and first mentions of oneapi date back 4 years ago, so I believe the support is ok for current versions.